de.metas.swat/de.metas.swat.base/src/main/sql/postgresql/system/40-sw01_swat/5452310_mig_FRESH-783_Migrate_Old_Inouts_async.sql

```diff
@@ -0,0 +1,167 @@
+
+
+
+CREATE OR REPLACE FUNCTION create_request_document_number ()
+RETURNS character varying
+AS
+$BODY$
+
+
+WITH updated AS (
+	UPDATE AD_Sequence 
+	SET CurrentNext = CurrentNext + 1
+	WHERE AD_Sequence_ID = 545441 -- this is the ID of the document number sequence. 
+	--I did not uset he name 'DocumentNo_R_Request' because there is another sequence with this name in the db and that one is incorrect
+	RETURNING (CurrentNext - 1) :: character varying as documentNo) 
+ select documentNo  FROM updated;
+
+$BODY$
+LANGUAGE sql volatile;
+
+
+
+
+DROP TABLE IF EXISTS temp_M_InOutLine_To_R_Request; 
+
+CREATE TABLE temp_M_InOutLine_To_R_Request
```

Comment:
To me it looks as if this `CREATE TABLE` actually does the main work, iterating those 1.8m inout lines metioned by mark. Therefore i think this does not solve the problem of the migration taking too long for the rollout window.

**Update:** i just checked at our biggest client, with the from, join and where clause from this select:
```
SELECT count(iol.*)
FROM M_InOutLine iol
 	JOIN M_InOut io on iol.M_InOut_ID = io.M_InOut_ID
 WHERE iol.QualityDiscountPercent > 0
 		and iol.IsActive = 'Y' 
 		and io.IsActive = 'Y'
  		and io.MovementDate >= '2015-01-01'
```
**Returns: 5679 rows.**

@metas-mk @metas-rc : please clarify how make iol we really need to migrate.